meeting-minutes/2023-06-13.md

```diff
@@ -0,0 +1,161 @@
+  - Dan Connolly (dckc) (Agoric)
+  - Ian Denhardt (zenhack) (Cap'n Proto)
+  - ICRainbow (icrainbow)
+  - Kris Kowal (kris) (Agoric)
+  - Richard Gibson (richard) (Agoric)
+  - Jim 
+  - David Thompson (dave) (Spritely)
+
+# [#57](https://github.com/ocapn/ocapn/issues/57) Agenda
+
+ - [#56](https://github.com/ocapn/ocapn/issues/56) Ordering conversation about serialization/etc
+ - [#6](https://github.com/ocapn/ocapn/issues/6#issuecomment-1589840247) Did DanC find the right API for the interop demo?
+ - [#8](https://github.com/ocapn/ocapn/issues/8) Jessica Tallon demoing test suite status
+ - [#53](https://github.com/ocapn/ocapn/issues/53) 3rd party handoff: should it be general or specific to certain netlayers?
+ 
+# Minutes
+
+markm: not as prepared as I'd like on ordering details #56
+
+cwebber: Jessica Tallon could show test suite
+
+## Implementation plans
+
+zenhack: perhaps folks who are about to implement something that might ossify should give a heads-up. nothing from capn-proto
+
+markm: an encodePassable PR is in review. distinguishing -0 from 0 comes up in this [PR](https://github.com/endojs/endo/pull/1626)
+
+## [#12](https://github.com/ocapn/ocapn/issues/12) [#53](https://github.com/ocapn/ocapn/issues/53) TLS+TCP netlayer, Third Party Handoffs (3PH)
+
+tsyesika: The TCP netlayer that Spritely has implemented hasn't changed how handoffs work. For the application we have used it for so far, we've been careful to avoid handoffs as it currently lacks any NAT holepunching, but we do plan to implement something like STUN. We haven't changed the 3PH implementation.
+
+cwebber: Ian's point that maybe there could be specicalized handoffs per netlayer is interesting.  is it a good idea?
+
+markm: 3PH is _mutually suspicious... core operator_ somewhat surprising that Agoric has been able to postpone it this far --
+ian: Cap'n Proto has likewise gone for a decade without it
+
+markm: Part of the goal of OCapN is to have a generalized mutually suspicious capability fabric regardless of particular network encoding, so having generalized handoffs useful there
+
+zenhack: I figured we'd have a protocol that might negotiate how to connect, eg a browser might say "I can connect over websockets and webrtc, but I can't connect over blockchain, I'm sorry".  Ideally Agoric and Spritely converge on a shared protocol but capn'proto probably won't, but I'd like to make it where if there's an OCapN handoff between two cap'n Proto vats that they can choose to cut out OCapN, and vice versa.  I think those are my clarifying goals
+
+dave: *describing the tls+tcp netlayer*
+
+kris: very interested in going down this rabbit hole, had a conversation about what non-blockchain protocol would be. We are considering a layering like “TCP + messaging + ed25519 encryption + capability transfer” without TLS, using an elliptic curve for signing and a different elliptic curve for transport opacity.
+
+dckc: Speaking of rabbit holes, point of order
```

Comment:
```suggestion
```

if we're not recording what the point of order was, this note seems out of place. (The point was to see where we were in the agenda and how to proceed without our chair.)